app/src/main/java/com/vero/tasky/agenda/data/repository/ReminderRepositoryImpl.kt

```diff
@@ -0,0 +1,74 @@
+package com.vero.tasky.agenda.data.repository
+
+import com.vero.tasky.agenda.data.local.dao.ModifiedAgendaItemDao
+import com.vero.tasky.agenda.data.local.dao.ReminderDao
+import com.vero.tasky.agenda.data.local.entities.ModifiedAgendaItemEntity
+import com.vero.tasky.agenda.data.mappers.*
+import com.vero.tasky.agenda.data.remote.network.api.ReminderApi
+import com.vero.tasky.agenda.domain.model.AgendaItem
+import com.vero.tasky.agenda.domain.model.AgendaItemType
+import com.vero.tasky.agenda.domain.model.ModificationType
+import com.vero.tasky.agenda.domain.remindermanager.AlarmData
+import com.vero.tasky.agenda.domain.remindermanager.AlarmHandler
+import com.vero.tasky.agenda.domain.repository.ReminderRepository
+import com.vero.tasky.core.data.remote.safeSuspendCall
+
+class ReminderRepositoryImpl(
+    private val api: ReminderApi,
+    private val reminderDao: ReminderDao,
+    private val modifiedAgendaItemDao: ModifiedAgendaItemDao,
+    private val alarmHandler: AlarmHandler,
+) : ReminderRepository {
+
+    override suspend fun saveReminder(
+        reminder: AgendaItem.Reminder,
+        modificationType: ModificationType
+    ): Result<Unit> {
+        alarmHandler.setAlarm(
+            AlarmData(
+                time = reminder.remindAt,
+                itemId = reminder.id,
+                title = reminder.title,
+                description = reminder.description,
+                type = AgendaItemType.REMINDER.toString()
+            )
+        )
+        reminderDao.insertReminders(reminder.toReminderEntity())
+        safeSuspendCall {
+            if (modificationType == ModificationType.CREATED)
+                api.createReminder(reminder.toReminderRequest())
+            else
+                api.updateReminder(reminder.toReminderRequest())
+        }.onFailure {
+                modifiedAgendaItemDao.insertAgendaItem(
+                    ModifiedAgendaItemEntity(
+                        id = reminder.id,
+                        agendaItemType = AgendaItemType.REMINDER,
+                        modificationType = modificationType
+                    )
+                )
+            }
+        return Result.success(Unit)
```

Comment:
If you are always returning a Success...
Maybe this whole function can just be a "void" and not return anything